src/contracts/core/DelegationManager.sol

```diff
@@ -264,8 +278,38 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         // actually undelegate the staker
         emit StakerUndelegated(staker, operator);
         delegatedTo[staker] = address(0);
+    }
 
-        return withdrawalRoot;
+    /**
+     * @notice Called by a staker to exit the "undelegation limbo" mode.
+     * @param middlewareTimesIndex Passed on as an input to the `slasher.canWithdraw` function, to ensure that the caller can exit undelegation limbo.
+     * This is because undelegation limbo is subject to some of the same restrictions as completing a withdrawal, to ensure that a staker cannot use undelegation
+     * limbo to avoid potentially being subject to slashing.
+     * @dev Note that this checks against the frozen status of the operator *that the caller was delegated to when they entered undelegation limbo*, to provide
+     * the same kind of slashing eligibility as withdrawals via the StrategyManager or DelegationManager.
+     */
+    function exitUndelegationLimbo(
+        uint256 middlewareTimesIndex
+    ) external onlyNotFrozen(_stakerUndelegationLimboStatus[msg.sender].delegatedAddress) {
```

Comment:
Is `onlyNotFrozen(msg.sender)` relevant here? That's what the current `EPM.exitUndelegationLimbo` has.

I think I'm becoming less of a fan of keeping in this slasher stuff - it makes me question its use during refactors, and the answer is inevitably something like "we don't know yet." But I worry that by leaving it in, we're going to become blind to its presence and not think about it hard enough when we eventually do add it.

Reply:
specifically it is the address that the Staker *was* delegated to being checked against here.
That being said, this appears to be redundant with the call to `Slasher.canWithdraw` below; I will remove it.
We'll need to revisit these calls regardless. I can understand your stance, and my own position feels increasingly harder to justify. Perhaps the best explanation I can offer is "this seems compliant with the (preliminary) interface of the Slasher, and ensures that the DelegationManager has the sufficient info to communicate here / perform the same check as the StrategyManager + EigenPodManager.

Reply:
removed the modifier in a commit https://github.com/Layr-Labs/eigenlayer-contracts/pull/208/commits/a3963f2c4f446269477e759e7c2332450917759a

Reply:
I'm happy for now knowing that it is in fact redundant with the call to `Slasher.canWithdraw`, and can be removed.

Is the current `onlyNotFrozen(msg.sender)` check completely unneeded in the EPM then?

Reply:
> Is the current onlyNotFrozen(msg.sender) check completely unneeded in the EPM then?

on completing a queued withdrawal? yes, seems redundant as well. likewise with the StrategyManager.

I guess the conundrum here is that the `Slasher.canWithdraw` function doesn't actually check frozen status of the operator -- so perhaps for now it actually makes sense to put this modifier back in place 😞

Reply:
I actually think leaving the modifier in is the sanest thing to do right now, so am reverting my previous commit which removed it.
For M2 testnet we are treating these as no-ops, so it shouldn't really matter either way, but I think what we should have here is consistency for now.
Going forward I think we should absorb this modifier's behavior into the `Slasher.canWithdraw` function (and further refine the behavior of the function), but I would prefer just leaving that off the table for this PR.
Change reverted here https://github.com/Layr-Labs/eigenlayer-contracts/pull/208/commits/c967c7441bcea58d9d836d4325a29deee59762af -- if we want to remove this then we should also do it in the `StrategyManager._completeQueuedWithdrawal` and `EigenPodManager._completeQueuedWithdrawal` functions as well, as part of a combined change.